packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -190,7 +190,7 @@ export const useGridColumnResize = (
     });
   };
 
-  const handleResizeMouseUp = useEventCallback((nativeEvent: MouseEvent) => {
```

Comment:
I can still reproduce the issue in https://codesandbox.io/s/9jbokp?file=/demo.tsx


https://github.com/mui/mui-x/assets/13808724/cddcfa18-41d3-4dc8-8630-1bf48bd5e4d2

Reply:
I've modified the implementation, instead of removing the click canceler in a microtask we're now doing it after a 100ms timeout. Although a timer isn't my favourite implementation, I couldn't find a way to reliably cancel the click otherwise.

Reply:
Maybe we can leverage `pointerEvents` here instead?
```js
colElementRef.current.style.pointerEvents = 'none';
```

Reply:
Yes, that works.

Reply:
Works when I test manually, fails when it runs with karma :neutral_face:

Reply:
I'm guessing it's because the testing `fireEvent.click()` doesn't respect `pointer-events: none` :neutral_face:

Reply:
And we can't really remove it because then what would we even be testing...

Reply:
@cherniavskii Let me know what you prefer. The way I see it we have either an optimal but untestable solution, or an unoptimal but testable one.

Reply:
We can cover it with an e2e test, I had a similar problem in https://github.com/mui/mui-x/pull/5808#discussion_r947782245

Reply:
Nice setup. E2E test did fail on master, and works on this branch. This is ready for review & merging.